<?xml version="1.0" encoding="UTF-8" ?>
<launch>

  <include file="$(find vrep4_helpers)/launch/kinect_pc.launch"/>
  <include file="$(find floor_nav)/launch/launch_vrep.launch"/>
	
  <!-- Traversability map node -->
  <node pkg="floor_plane_mapping" type="traversability_occgrid" name="traversability_occgrid" output="screen">
      launch-prefix="xterm -e gdb --args"
      <param name="base_frame" value="/world" />
      <param name="robot_frame" value="/bubbleRob" />
      <param name="max_range" value="3.0" />
      <param name="grid_size" value="0.05" />
      <param name="beta" value="100" />
      <param name="gama" value="10" />

      <remap from="/traversability_occgrid/scans" to="/depth_registered/points"/>
  </node>

   
  <!-- Cylinder detector node -->
  <node pkg="tf" type="static_transform_publisher" name="test_new_tf" args="0 0 0 0 -1.5708 1.5708 camera_color_optical_frame child_camera_color_optical_frame 100" />
  <node pkg="cylinder_detection" type="cylinder_detection" name="cylinder_detection" output="screen">
      launch-prefix="xterm -e gdb --args"
      <param name="base_frame" value="/world" />
      <param name="robot_frame" value="/bubbleRob" />
      <param name="max_range" value="2.0" />
      <param name="n_samples" value="8000" />

      <param name="tolerance" value="0.004" />
      <remap from="/cylinder_detection/scans" to="/depth_registered/points"/>
  </node>


  <!-- Rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(floor_plane_mapping)/project1.rviz" />

  <!-- image_view -->
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/occupancy_grid" />
  </node>

  <node name="test_graph" pkg="floor_nav" type="test_graph.py" output="screen">
  </node>

</launch>
